bindings/python/pymongoarrow/bson.pyi

```diff
@@ -0,0 +1,66 @@
+# Copyright 2021-present MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def process_bson_stream(bson_stream, context):
+    cdef const uint8_t* docstream = <const uint8_t *>bson_stream
+    cdef size_t length = <size_t>PyBytes_Size(bson_stream)
+    cdef bson_reader_t* stream_reader = bson_reader_new_from_data(docstream, length)
+    cdef cbool reached_eof = False
+    cdef const bson_t * doc = bson_reader_read(stream_reader, & reached_eof)
+    cdef bson_iter_t doc_iter
+    cdef const char* key
+    cdef bson_type_t value_t
+
+    builder_map = context.builder_map
+    type_map = context.type_map
+
+    while doc != NULL:
+        if not bson_iter_init(&doc_iter, doc):
+            raise RuntimeError
+        while bson_iter_next(&doc_iter):
+            key = bson_iter_key(&doc_iter)
+            if key in builder_map:
+                builder = builder_map[key]
+                ftype = type_map[key]
+                value_t = bson_iter_type(&doc_iter)
+                if ftype == _BsonArrowTypes.int32:
+                    if value_t == BSON_TYPE_INT32:
```

Comment:
Should int32 also accept BSON_TYPE_BOOL like int64 and double? There should be tests for each of these conversion rules. 

Will these lossy conversion rules cause problems when we later add support for writing (BSON bool -> int64 -> BSON int64)?

Reply:
Oh I see, the issue is that libbson provides bson_iter_as_int64 and bson_iter_as_double which does the casting. There is no bson_iter_as_int32 for us to use here so this code SGTM as is.